File: Leetcode/python/sol897.py

# https://leetcode.com/problems/increasing-order-search-tree/description/

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def increasingBST(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        temp = TreeNode(val=-1)
        dummy = temp
        def helper(node):
            nonlocal dummy
            if not node:
                return
            helper(node.left)
            dummy.right = TreeNode(val=node.val)
            # node.left = None
            # dummy.right = node
            dummy = dummy.right
            helper(node.right)
        helper(root)
        return temp.right

File: Leetcode/typescript/sol897.ts

//https://leetcode.com/problems/increasing-order-search-tree/description/

/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function increasingBST(root: TreeNode | null): TreeNode | null {
    let dummy = new TreeNode(-1);
    let temp = dummy
    const helper = (node) => {
        if(!node){
            return;
        }
        helper(node.left)
        temp.right = new TreeNode(node.val)
        temp = temp.right
        helper(node.right)
    }
    helper(root)
    return dummy.right;
};